bool detectLeft
float32 angleLeft
float32 distanceLeft
bool detectRight
float32 angleRight
float32 distanceRight
bool detectTop
float32 angleTop
float32 distanceTop